src/lib/libc_preload/libc_impls.c

```diff
@@ -195,3 +196,41 @@ int __fxstat64(int ver, int a, struct stat64* b) {
     _convert_stat_to_stat64(&s, b);
     return rv;
 }
+
+int open(const char* pathname, int flags, ...) {
+    va_list(args);
+    va_start(args, flags);
+    mode_t mode = va_arg(args, mode_t);
+    va_end(args);
+
+    // "If neither O_CREAT nor O_TMPFILE is specified in flags, then mode is
+    // ignored"
+    //
+    // We explicitly set to 0 here so that strace logging doesn't log an
+    // arbitrary value for `mode` when it wasn't explicitly provided by the
+    // caller.
+    if (!(flags & (O_CREAT | O_TMPFILE))) {
+        mode = 0;
+    }
```

Comment:
It looks like this is what glibc does as well:

https://github.com/bminor/glibc/blob/b92a49359f33a461db080a33940d73f47c756126/sysdeps/unix/sysv/linux/open.c#L30-L41

```c
int mode = 0;

if (__OPEN_NEEDS_MODE (oflag))
{
  va_list arg;
  va_start (arg, oflag);
  mode = va_arg (arg, int);
  va_end (arg);
}
```